### PR TITLE
[risk=low][RW-12400] Convert 2 Enzyme tests to RTL

### PR DESCRIPTION
**Pull request by jmthibault79**
---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/with-leo-cookie.spec.tsx [hunk 1]
```diff
@@ -153,15 +147,14 @@ describe(withLeoCookie.name, () => {
       .spyOn(leoProxyApi(), 'setCookie')
       .mockResolvedValue(null);
 
-    const wrapper = createWrapper();
+    const { unmount } = createWrapper();
 
     expect(setCookieStub).toHaveBeenCalledTimes(1);
 
-    wrapper.unmount();
+    unmount();
```
Comment (Qi77Qi) [inline ui/src/app/components/with-leo-cookie.spec.tsx:154]: I could be wrong...but looks like it needs to be `unmount.unmount()`?

Comment (jmthibault79) [reply ui/src/app/components/with-leo-cookie.spec.tsx:154]: I don't know much about what it's doing here, but why do you think that?

Comment (evrii) [reply ui/src/app/components/with-leo-cookie.spec.tsx:154]: I think this looks good. It looks like Joel is destructuring the unmount function from the return of render.

A little more detail about that can be found here:
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-wrapper-as-the-variable-name-for-the-return-value-from-render

File: ui/src/app/components/with-new-user-satisfaction-survey-modal.spec.tsx [hunk 2]
```diff
@@ -91,30 +85,39 @@ describe('withNewUserSatisfactionSurveyModal', () => {
...
       .mockImplementationOnce(() => Promise.reject());
-
     expect(notificationStore.get()).toBeNull();
-
     await createWrapperAtPath(`?surveyCode=${code}`);
-
-    expect(notificationStore.get().title).toBeTruthy();
-    expect(notificationStore.get().message).toBeTruthy();
+    await waitFor(() => expect(notificationStore.get().title).toBeTruthy());
+    await waitFor(() => expect(notificationStore.get().message).toBeTruthy());
```
Comment (evrii) [inline ui/src/app/components/with-new-user-satisfaction-survey-modal.spec.tsx:91]: I am not sure if this is the case or not, but if title and message attributes are being set at the same time, you might be able to get away on just the first statement having a waitfor. I don't think it is a big deal either way though.

**Review by evrii — APPROVED** [review]

**Pull request merged**